### ISSUE TITLE
_renpysteam functions result on TypeError

### ISSUE THREAD

**Issue by raspberry-soft** [renpy/renpy#3861]
While having `config.steam_appid` set and steam opened, using this:

```
$ subscribed_items = _renpysteam.get_subscribed_items()
text "subscribed_items = [subscribed_items]"
```

Reports this error:

```
I'm sorry, but an uncaught exception occurred.

While running game code:
  File "game/scripts/system/debug/debug_script.rpy", line 46, in execute
    screen debug_script():
  File "game/scripts/system/debug/debug_script.rpy", line 46, in execute
    screen debug_script():
  File "game/scripts/system/debug/debug_script.rpy", line 58, in execute
    drag:
  File "game/scripts/system/debug/debug_script.rpy", line 63, in execute
    frame:
  File "game/scripts/system/debug/debug_script.rpy", line 64, in execute
    has vbox
  File "game/scripts/system/debug/debug_script.rpy", line 88, in execute
    showif page == "images":
  File "game/scripts/system/debug/debug_script.rpy", line 721, in execute
    if achievement.steamapi:
  File "game/scripts/system/debug/debug_script.rpy", line 804, in execute
    $ subscribed_items = _renpysteam.get_subscribed_items()  # TODO: REPORT BUG
  File "game/scripts/system/debug/debug_script.rpy", line 804, in <module>
    $ subscribed_items = _renpysteam.get_subscribed_items()  # TODO: REPORT BUG
  File "renpy/common/00steam.rpy", line 471, in get_subscribed_items
    count = steamapi.SteamUGC().GetSubscribedItems(byref(subscribed), 512)
ArgumentError: argument 2: <class 'TypeError'>: expected LP_c_ulonglong instance instead of pointer to c_ulonglong_Array_512

-- Full Traceback ------------------------------------------------------------

Full traceback:
  File "renpy/common/_layout/screen_main_menu.rpym", line 28, in script
    python hide:
  File "C:\Users\***\Desktop\My Files\Software\renpy-7.4.11-sdk\renpy\ast.py", line 1111, in execute
    renpy.python.py_exec_bytecode(self.code.bytecode, self.hide, store=self.store)
  File "C:\Users\***\Desktop\My Files\Software\renpy-7.4.11-sdk\renpy\python.py", line 1049, in py_exec_bytecode
    exec(bytecode, globals, locals)
  File "renpy/common/_layout/screen_main_menu.rpym", line 28, in <module>
    python hide:
  File "renpy/common/_layout/screen_main_menu.rpym", line 35, in _execute_python_hide
    ui.interact()
  File "C:\Users\***\Desktop\My Files\Software\renpy-7.4.11-sdk\renpy\ui.py", line 299, in interact
    rv = renpy.game.interface.interact(roll_forward=roll_forward, **kwargs)
  File "C:\Users\***\Desktop\My Files\Software\renpy-7.4.11-sdk\renpy\display\core.py", line 3376, in interact
    repeat, rv = self.interact_core(preloads=preloads, trans_pause=trans_pause, pause=pause, pause_start=pause_start, **kwargs) # type: ignore
  File "C:\Users\***\Desktop\My Files\Software\renpy-7.4.11-sdk\renpy\display\core.py", line 3803, in interact_core
    root_widget.visit_all(lambda i : i.per_interact())
  File "C:\Users\***\Desktop\My Files\Software\renpy-7.4.11-sdk\renpy\display\core.py", line 582, in visit_all
    d.visit_all(callback, seen)
  File "C:\Users\***\Desktop\My Files\Software\renpy-7.4.11-sdk\renpy\display\core.py", line 582, in visit_all
    d.visit_all(callback, seen)
  File "C:\Users\***\Desktop\My Files\Software\renpy-7.4.11-sdk\renpy\display\core.py", line 582, in visit_all
    d.visit_all(callback, seen)
  [Previous line repeated 1 more time]
  File "C:\Users\***\Desktop\My Files\Software\renpy-7.4.11-sdk\renpy\display\screen.py", line 451, in visit_all
    callback(self)
  File "C:\Users\***\Desktop\My Files\Software\renpy-7.4.11-sdk\renpy\display\core.py", line 3803, in <lambda>
    root_widget.visit_all(lambda i : i.per_interact())
  File "C:\Users\***\Desktop\My Files\Software\renpy-7.4.11-sdk\renpy\display\screen.py", line 462, in per_interact
    self.update()
  File "C:\Users\***\Desktop\My Files\Software\renpy-7.4.11-sdk\renpy\display\screen.py", line 653, in update
    self.screen.function(**self.scope)
  File "game/scripts/system/debug/debug_script.rpy", line 46, in execute
    screen debug_script():
  File "game/scripts/system/debug/debug_script.rpy", line 46, in execute
    screen debug_script():
  File "game/scripts/system/debug/debug_script.rpy", line 58, in execute
    drag:
  File "game/scripts/system/debug/debug_script.rpy", line 63, in execute
    frame:
  File "game/scripts/system/debug/debug_script.rpy", line 64, in execute
    has vbox
  File "game/scripts/system/debug/debug_script.rpy", line 88, in execute
    showif page == "images":
  File "game/scripts/system/debug/debug_script.rpy", line 721, in execute
    if achievement.steamapi:
  File "game/scripts/system/debug/debug_script.rpy", line 804, in execute
    $ subscribed_items = _renpysteam.get_subscribed_items()  # TODO: REPORT BUG
  File "game/scripts/system/debug/debug_script.rpy", line 804, in <module>
    $ subscribed_items = _renpysteam.get_subscribed_items()  # TODO: REPORT BUG
  File "renpy/common/00steam.rpy", line 471, in get_subscribed_items
    count = steamapi.SteamUGC().GetSubscribedItems(byref(subscribed), 512)
  File "/home/tom/ab/renpy-build/tmp/py3/steam/steamapi.py", line 6855, in GetSubscribedItems
ArgumentError: argument 2: <class 'TypeError'>: expected LP_c_ulonglong instance instead of pointer to c_ulonglong_Array_512

Windows-10-10.0.19041 AMD64
Ren'Py 8.0.0.22062402
Thu Aug 11 18:31:16 2022

```

Also, while using:

```
$ beta_name = _renpysteam.get_current_beta_name()
text "beta_name = [beta_name]"
```

It should report None, but instead:

```
I'm sorry, but an uncaught exception occurred.

While running game code:
  File "game/scripts/system/debug/debug_script.rpy", line 46, in execute
    screen debug_script():
  File "game/scripts/system/debug/debug_script.rpy", line 46, in execute
    screen debug_script():
  File "game/scripts/system/debug/debug_script.rpy", line 58, in execute
    drag:
  File "game/scripts/system/debug/debug_script.rpy", line 63, in execute
    frame:
  File "game/scripts/system/debug/debug_script.rpy", line 64, in execute
    has vbox
  File "game/scripts/system/debug/debug_script.rpy", line 88, in execute
    showif page == "images":
  File "game/scripts/system/debug/debug_script.rpy", line 721, in execute
    if achievement.steamapi:
  File "game/scripts/system/debug/debug_script.rpy", line 778, in execute
    $ beta_name = _renpysteam.get_current_beta_name()  # TODO: REPORT BUG
  File "game/scripts/system/debug/debug_script.rpy", line 778, in <module>
    $ beta_name = _renpysteam.get_current_beta_name()  # TODO: REPORT BUG
  File "renpy/common/00steam.rpy", line 228, in get_current_beta_name
    if not steamapi.SteamApps().GetCurrentBetaName(byref(rv), 256):
ArgumentError: argument 2: <class 'TypeError'>: wrong type

-- Full Traceback ------------------------------------------------------------

Full traceback:
  File "renpy/common/_layout/screen_main_menu.rpym", line 28, in script
    python hide:
  File "C:\Users\***\Desktop\My Files\Software\renpy-7.4.11-sdk\renpy\ast.py", line 1111, in execute
    renpy.python.py_exec_bytecode(self.code.bytecode, self.hide, store=self.store)
  File "C:\Users\***\Desktop\My Files\Software\renpy-7.4.11-sdk\renpy\python.py", line 1049, in py_exec_bytecode
    exec(bytecode, globals, locals)
  File "renpy/common/_layout/screen_main_menu.rpym", line 28, in <module>
    python hide:
  File "renpy/common/_layout/screen_main_menu.rpym", line 35, in _execute_python_hide
    ui.interact()
  File "C:\Users\***\Desktop\My Files\Software\renpy-7.4.11-sdk\renpy\ui.py", line 299, in interact
    rv = renpy.game.interface.interact(roll_forward=roll_forward, **kwargs)
  File "C:\Users\***\Desktop\My Files\Software\renpy-7.4.11-sdk\renpy\display\core.py", line 3376, in interact
    repeat, rv = self.interact_core(preloads=preloads, trans_pause=trans_pause, pause=pause, pause_start=pause_start, **kwargs) # type: ignore
  File "C:\Users\***\Desktop\My Files\Software\renpy-7.4.11-sdk\renpy\display\core.py", line 3803, in interact_core
    root_widget.visit_all(lambda i : i.per_interact())
  File "C:\Users\***\Desktop\My Files\Software\renpy-7.4.11-sdk\renpy\display\core.py", line 582, in visit_all
    d.visit_all(callback, seen)
  File "C:\Users\***\Desktop\My Files\Software\renpy-7.4.11-sdk\renpy\display\core.py", line 582, in visit_all
    d.visit_all(callback, seen)
  File "C:\Users\***\Desktop\My Files\Software\renpy-7.4.11-sdk\renpy\display\core.py", line 582, in visit_all
    d.visit_all(callback, seen)
  [Previous line repeated 1 more time]
  File "C:\Users\***\Desktop\My Files\Software\renpy-7.4.11-sdk\renpy\display\screen.py", line 451, in visit_all
    callback(self)
  File "C:\Users\***\Desktop\My Files\Software\renpy-7.4.11-sdk\renpy\display\core.py", line 3803, in <lambda>
    root_widget.visit_all(lambda i : i.per_interact())
  File "C:\Users\***\Desktop\My Files\Software\renpy-7.4.11-sdk\renpy\display\screen.py", line 462, in per_interact
    self.update()
  File "C:\Users\***\Desktop\My Files\Software\renpy-7.4.11-sdk\renpy\display\screen.py", line 653, in update
    self.screen.function(**self.scope)
  File "game/scripts/system/debug/debug_script.rpy", line 46, in execute
    screen debug_script():
  File "game/scripts/system/debug/debug_script.rpy", line 46, in execute
    screen debug_script():
  File "game/scripts/system/debug/debug_script.rpy", line 58, in execute
    drag:
  File "game/scripts/system/debug/debug_script.rpy", line 63, in execute
    frame:
  File "game/scripts/system/debug/debug_script.rpy", line 64, in execute
    has vbox
  File "game/scripts/system/debug/debug_script.rpy", line 88, in execute
    showif page == "images":
  File "game/scripts/system/debug/debug_script.rpy", line 721, in execute
    if achievement.steamapi:
  File "game/scripts/system/debug/debug_script.rpy", line 778, in execute
    $ beta_name = _renpysteam.get_current_beta_name()  # TODO: REPORT BUG
  File "game/scripts/system/debug/debug_script.rpy", line 778, in <module>
    $ beta_name = _renpysteam.get_current_beta_name()  # TODO: REPORT BUG
  File "renpy/common/00steam.rpy", line 228, in get_current_beta_name
    if not steamapi.SteamApps().GetCurrentBetaName(byref(rv), 256):
  File "/home/tom/ab/renpy-build/tmp/py3/steam/steamapi.py", line 6035, in GetCurrentBetaName
ArgumentError: argument 2: <class 'TypeError'>: wrong type

Windows-10-10.0.19041 AMD64
Ren'Py 8.0.0.22062402
Thu Aug 11 18:37:09 2022
```

EDIT:

There's a typo in the doc string of the following function:

achievements should be achievement.

```
    def activate_overlay_to_store(appid, flag=None):
        """
        :doc: steam_overlay

        `flag`
            One of achievements.steam.STORE_NONE, .STORE_ADD_TO_CART, or .STORE_ADD_TO_CART_AND_SHOW.
        """

```

Many other _renpysteam functions I tested work fine.

**Comment by renpytom** [comment]
I've exposed the entire steam api, as achievement.steamapi, so you can use that directly. I don't plan to expose individual functions, beyond what I've already done, unless Ren'Py uses those functions itself.

**Comment by renpytom** [comment]
Closed by a45c39b5ef48e834a972dbe3680f13b87aef232e .
